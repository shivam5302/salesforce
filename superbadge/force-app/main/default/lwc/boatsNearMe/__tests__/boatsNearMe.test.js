import { createElement } from 'lwc';
import BoatsNearMe from 'c/boatsNearMe';
import getBoatsByLocation from '@salesforce/apex/BoatDataService.getBoatsByLocation';
import getBoatsByLocatoinMockData from './data/getBoatsByLocation.json'

jest.mock('@salesforce/apex/BoatDataService.getBoatsByLocation',
	() => {
		const { createApexTestWireAdapter } = require('@salesforce/sfdx-lwc-jest');
		return {
			default: createApexTestWireAdapter(jest.fn())
		};
	},
	{ virtual: true }
)

const flushPromises = async () => new Promise((resolve) => setTimeout(resolve, 0));

describe('c-boats-near-me', () => {
	let element;
	beforeEach(()=>{
		element = createElement('c-boats-near-me', {
			is: BoatsNearMe
		});
		document.body.appendChild(element);

		global.navigator.geolocation = {
			getCurrentPosition: jest.fn()
		};
	})
	afterEach(() => {
		// The jsdom instance is shared across test cases in a single file so reset the DOM
		while (document.body.firstChild) {
			document.body.removeChild(document.body.firstChild);
		}
	});

	it('Should render the spinner', async() => {
		expect(element).toMatchSnapshot();
	});

	it('Should render the boats by location', async() => {
		//Given
		
		//When
		getBoatsByLocation.emit(getBoatsByLocatoinMockData);
		
		//Then
		await flushPromises();
		const lightningMap = element.shadowRoot.querySelector('lightning-map');
		expect(lightningMap.mapMarkers.length).toBe(2);
	});

	// it('Should render the boats by location', () => {

	// 	const mockPosition = { coords: { latitude: 40.7128, longitude: -74.0060 } };
		
	// 	// Mock the getCurrentPosition to invoke the success callback
	// 	navigator.geolocation.getCurrentPosition.mockImplementationOnce((success) => {
	// 		success(mockPosition);
	// 	});
	// 	getBoatsByLocation.emit()

	// 	// Assert
	// 	// const div = element.shadowRoot.querySelector('div');
	// 	expect(1).toBe(1);
	// });
});