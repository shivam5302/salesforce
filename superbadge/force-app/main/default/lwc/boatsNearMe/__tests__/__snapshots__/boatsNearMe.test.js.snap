// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`c-boats-near-me Should render the spinner 1`] = `
<c-boats-near-me>
  #shadow-root(open)
    <lightning-card
      class="slds-is-relative"
    >
      #shadow-root(open)
        <slot />
        <slot
          name="actions"
        />
        <slot
          name="footer"
        />
        <slot
          name="title"
        />
      <lightning-spinner>
        #shadow-root(open)
      </lightning-spinner>
      <div
        slot="footer"
      >
        Top 10 Only!
      </div>
    </lightning-card>
</c-boats-near-me>
`;
